frappe/website/page_renderers/static_page.py

```diff
@@ -31,7 +31,7 @@ def can_render(self):
 		return self.is_valid_file_path() and self.file_path
 
 	def is_valid_file_path(self):
-		extension = self.path.rsplit(".", 1)[-1]
+		extension = self.path.rsplit(".", 1)[-1] if "." in self.path else ""
```

Comment:
Can you share one example of `self.path` where this is useful?

Reply:
Hi @ankush, @barredterra is right. There are cases where there is no extension. Extension is extracted doing a dot rsplit, but some times files have no extensions and are valid as static files. Below line only checks if extension is not within this list, to consider it as valid file path:
```
UNSUPPORTED_STATIC_PAGE_TYPES = ("html", "md", "js", "xml", "css", "txt", "py", "json")
```

Reply:
@developmentforpeople then the change you made doesn't make any difference? 

in case there's no extension `"string".rsplit(".", 1)[-1]` will just return same string, but now it will return an empty string... either way it's not passing the check?

Reply:
The code we are talking about is extracting the file extension and assigning it to a "extension" variable and then the extension is contrasted with a list of valid ones:
```
		extension = self.path.rsplit(".", 1)[-1] if "." in self.path else ""
		if extension in UNSUPPORTED_STATIC_PAGE_TYPES:
			return False
		return True
```
@ankush you are right, the code in both cases will continue to work, but it is clearer if the code that extracts the "extension", do its work. (and if the file has no extension, it should be an empty string). You are right that both ways the code work, I am not fixing anything, but the code should be comprehensible and do what expected with variable names and checks :-).  This is my last try with this pull request. Thank you very much for your time :)

Reply:
@developmentforpeople ah that makes sense. I was just wondering in what case this actually breaks.